### ISSUE TITLE
Helmert operation ignores +t_obs

### ISSUE THREAD

**Issue by keyos89** [OSGeo/PROJ#1239]
According to the documentation:

> `+t_obs=<value>`
> Observation time of coordinate(s) given in decimalyear. Mostly useful in 2D and 3D transformations where the observation time is not passed as part of the input coordinate. **Can be used to override the observation time from the input coordinate.**

https://proj4.org/operations/transformations/helmert.html#cmdoption-arg-t-obs

However, it seems that the `+t_obs` value does not override the observation time from the input coordinate:

```
$ echo 10 60 0 2000 | cct -vvv +proj=pipeline +ellps=GRS80 +step +proj=cart +step +init=ITRF2000:ITRF2005 +inv +t_obs=2100
...
Final: proj=pipeline ellps=GRS80 step proj=cart step init=ITRF2000:ITRF2005 inv t_obs=2100 argc=10 pargc=8
Transformation parameters for observation t_obs=2000 (t_epoch=2000):
...
 3148533.3858    555171.3848  5500477.1302     2000.0000
```

Note that the debug information says `t_obs=2000` and not 2100 as was sent in as the `+t_obs` parameter. Changing the value of `+t_obs` confirms that it is ignored as the final answer is unchanged:

```
$ echo 10 60 0 2000 | cct -vvv +proj=pipeline +ellps=GRS80 +step +proj=cart +step +init=ITRF2000:ITRF2005 +inv +t_obs=1900
...
 3148533.3858    555171.3848  5500477.1302     2000.0000
```

Please advise if there is something we have misunderstood :blush:

**Comment by kbevers** [comment]
The `+t_obs` parameter was originally introduced to allow the use of 14 parameter shifts using entry points from the old `proj_api.h` API. The idea was to let the user specify the observation time as part of the transformation setup via `+t_obs`. As far as I can tell this has actually never worked due to flawed logic internally in the Helmert operation.

Since 4D coordinates are now more mature in PROJ I think the best way forward is to completely remove any trace of `+t_obs` from the Helmert operation. From version 6, `cs2cs` uses 4D coordinates and `proj` is restricted to projections so there's no apparent need for setting the observation time as part of the transformation setup any more. At least within the PROJ ecosphere. Consumers of the PROJ library need to use the `proj.h` API to gain access to spatiotemporal transformations any way so that should cause any problems either.

So, to sum up I will remove the `+t_obs` parameter from the Helmert operation and add a section in the docs that clarify how PROJ treats the temporal components of 4D coordinates.

I hope this will make life easier for you, @keyos89! Thanks for reporting this problem.

**Comment by busstoptaktik** [comment]
Also, @keyos89, note that `cct` supports the `-z` and `-t` command line options, allowing you to ignore the actual values for z and t provided in the input data.

This will do essentially what you use the (as @kbevers notes) flawed `+t_obs` transformation parameter for.
